### PR TITLE
add contributor ladder

### PR DESCRIPTION
**Pull request by krancour**
This new contributor ladder tentatively expands, revises, and replaces https://github.com/brigadecore/community/blob/main/governance.md

This is based heavily on work @carolynvs has done in TAG Contributor Strategy, and revised a good deal to suit the needs of the project.

Note that it was a deliberate choice to refrain from quantifying phrases like "several contributions" or "regularly contributing." I felt that at this stage of the project, leaving these subjective gives the project maintainers and core maintainers greater latitude and discretion in moving contributors up and down the ladder. There is no reason that cannot be revised in the future as the community grows and its needs change.

If this is somewhat lacking on details relating to code review processes, that is because those details will be covered in a separate doc, again, based on some of @carolynvs' excellent work in TAG Contributor Strategy.

In addition to @vdice, I would be eager for feedback on this PR from @carolynvs and @karenhchu.

### CODE REVIEW

File: docs/content/topics/contributor-guide/ladder.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+* Have earned the trust of the Core Maintainers
+* Exercise sound judgement for the good of their project(s) and the Project,
+  independent of their employer's interests
+* Contribute regularly to their project(s)
+* Conduct PR reviews
+* Mentor other Contributors
+* Regularly participate in Community meetings
+
+Privileges include:
```
Comment (carolynvs) [inline docs/content/topics/contributor-guide/ladder.md:94]: Do you want to mention that maintainers/core maintainers have write access to the repository/org?

**Review by carolynvs — APPROVED** [review]
LGTM!

File: docs/content/topics/contributor-guide/ladder.md (hunk 1 above)
Comment (krancour) [reply docs/content/topics/contributor-guide/ladder.md:94]: Yes. That's conspicuously absent. Thanks for catching that.

File: docs/content/topics/contributor-guide/ladder.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+* Publicly representing the Brigade Project
+* Communicating with CNCF on behalf of the Project
+* Stewardship of Project secrets (passwords, keys, etc.)
+* Having a vote in matters affecting Project direction, including the election
+  of new Project Maintainers and Core Maintainers
+
+Becoming a Core Maintainer requires a formal nomination
+by an existing Core Maintainer. All
+existing Core Maintainers are all eligible to vote.
```
Comment (vdice) [inline docs/content/topics/contributor-guide/ladder.md:137]:
```suggestion
existing Core Maintainers are eligible to vote.
```

**Review by vdice — APPROVED** [review]
One optional grammar suggestion, otherwise LGTM!

**Pull request merged**